packages/library-legacy/package.json

```diff
@@ -43,7 +43,7 @@
     "build:fixtures": "set -ex; ./test/fixtures/noop-program/build.sh",
     "clean": "rimraf ./doc ./declarations ./lib",
     "dev": "cross-env NODE_ENV=development rollup -c --watch",
-    "publish-packages": "semantic-release --repository-url git@github.com:solana-labs/solana-web3.js.git",
+    "publish-packages": "npm view $npm_package_name@$npm_package_version > /dev/null 2>&1 || semantic-release --repository-url git@github.com:solana-labs/solana-web3.js.git",
```

Comment:
Question: does this look right for library-legacy? I'm guessing `semantic-release ...` is kinda the same as `pnpm publish ...`, so I can do the same for it?

Reply:
I think this is OK. `semantic-release` will handle incrementing the packages versions beyond just the commit hash, but if the commit message doesn't warrant a new patch/minor/major, then this will perform the intended check.